### DISCUSSION TITLE
My wifi has disappeared on Klipper Mod v00.04-beta..

### DISCUSSION THREAD

**Discussion by Giterdun2** [xblax/flashforge_ad5m_klipper_mod#128] [Q&A]
Not sure what's going on, but for some reason I cannot see anything on my network wifi screen when selected in KlipperScreen.. it's totally blank.. reboots do nothing.. yet I can ping the ip, but cannot connect with Mainsail.. I rebooted back to factory to check and test the wifi and it's fine there.. I have the wifi set to a static ip as per instructions from another post.. and it has been working fine until now.. nothing was changed that I know of..  Anyone have any ideas on this?

**Comment by consp** [comment]
The code for managing the wifi networks changed and should have been more robust, apparently it isn't.

Could you open the wifi screen and afterwards send the log in `/var/log/KlipperScreen.log` here? (note: might contain wifi names so be careful what you post if this is an issue)

**Reply by Giterdun2** [reply]
I initially set up my wifi from THEGxHOxST instructions in another post here, as I wanted a static address.. and I thought it had worked, but apparently didn't as when I turned on my printer today, I couldn't connect.. I then found that it had another address thru my router info..  When I used SolarPutty to have a look and I found that my "wifi".psk file was gone, So I connected an ethernet cable to connect. I can putty into the printer, and located the file, but not very much linux savvy.. so not sure how to copy the klipperscreen.log file back to windows10..

**Reply by consp** [reply]
>  "wifi".psk file was gone

Had this once as well, @xblax shoudn't the wifi dir be preserved?

**Reply by Giterdun2** [reply]
This is what I was asked to do by THEGxHOxST for a static ip.. and as stated, I thought it was good, until it ended up with another ip.. 

cd /var/lib/iwd
ls
nano "yourwifi".psk
add to the bottom of that file.


[IPv4]
Address=xxx.xxx.xxx.xxx
Netmask=255.255.255.0
Gateway=xxx.xxx.xxx.254
Broadcast=your gateways subnet xxx.xxx.xxx.255
DNS=1.1.1.1

**Reply by Giterdun2** [reply]
hmm.. No matter what I tried, I could not get my wifi to work, so I resorted to uninstalling the mod, and then reinstalling it again, and now the wifi is working again. Not sure if something was corrupted or what. This time I set up the etho and wlan0 as static in the interfaces file with separate IPs as per thee info below, There's no option to turn wifi off when using ethernet, so what I did notice is that if ethernet is connected, the Wifi symbol on the screen  displays the ethernet address on the screen, even though it actually has a different ip address.  Now I can connect with either wifi or ethernet, and with both ips..
Used SolarPutty to edit... 

replace the x's with your subnet...

nano /etc/network/interfaces

replace

auto eth0
iface eth0 inet dhcp
pre-up /etc/network/nfs_check
wait-delay 15
hostname $(hostname)

With

auto eth0
iface eth0 inet static
address xxx.xxx.xxx.xxx(Enter desired ip here)
netmask 255.255.255.0
network xxx.xxx.xxx.0
broadcast xxx.xxx.xxx.255
gateway xxx.xxx.1.1
pre-up /etc/network/nfs_check
wait-delay 15
hostname $(hostname)

**Reply by consp** [reply]
> hmm.. No matter what I tried, I could not get my wifi to work,

Edited the files in windows by any chance? (if you used the usb method)

**Comment by Giterdun2** [comment]
> > hmm.. No matter what I tried, I could not get my wifi to work,
> 
> Edited the files in windows by any chance? (if you used the usb method)
I use Windows 10, but I used SolarPutty and edited in that as posted with nano.. In any case, I had my printer off for a couple days, and just found out that this didn't work either. It's kinda screwy.. It shows the correct IP address on the screen, but it was still assigned a different ip... so I decided to just go with the DHCP Reservation method in the router itself to assign a static address to the ethernet and wifi Mac addresses, and just use the ethernet for now..... Also getting Moonraker connection issues with 127.0.0.1 port 7125 so will be doing another uninstall and reinstall when time permits..

**Reply by consp** [reply]
Interesting that it still shows the old ip, looks like something is using the old settings. There is something funky going on with the IP's though, as if you use ethernet and wifi together you will only get the ethernet ip.

Tnx for the info, could be useful in figuring out what's wrong.

**Reply by Giterdun2** [reply]
Ya, it was really kinda wierd. It shows both the ethernet and the wifi with the same ip, which is the ethernet ip as you say...  The issue is that once you connect with wifi, which I did, I didn't know how or where to turn it off..  As mentioned, I uninstalled and reinstalled the KlipperMod and just left my wifi alone this time, although this time I did set the wifi and ethernet macs to static IPs in the router. For now, I'll just leave it alone until the next release and maybe play with it some more at that time... ;)
